components/tyk-dev-portal/templates/job-portal.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: dev-portal-job-{{ include "tyk-dev-portal.fullname" . }}
+  labels:
+    app: {{ include "tyk-dev-portal.fullname" . }}
+    chart: {{ include "tyk-dev-portal.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}  
+spec:
+  template:
+    spec:
+      containers:
+      - name: bootstrap-portal
+        image: curlimages/curl:8.8.0
+        env:
+        - name: ADMIN_EMAIL
+          value: "{{ .Values.global.adminUser.email  }}"
+        - name: ADMIN_PASSWORD
+          {{ if .Values.global.adminUser.useSecretName }}
+          valueFrom:
+            secretKeyRef:
+              name: secrets-{{ include "tyk-dev-portal.fullname" . }}
+              key: adminUserPassword
+          {{ else }}
+          value: {{ .Values.global.adminUser.password | quote }}
+          {{ end }}
+        - name: FIRSTNAME
+          value: "{{ .Values.bootstrapPortal.firstName }}"
+        - name: LASTNAME
+          value: "{{ .Values.bootstrapPortal.lastName }}"                    
+        - name: SERVICE_NAME
+          value: "dev-portal-svc-{{ include "tyk-dev-portal.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local"
+        - name: PORT
+          value: "{{ .Values.containerPort }}"   
+        command: 
+          - sh
+          - -c
+          - |
+            echo "Waiting for dev-portal to be ready..."
+            while ! nc -z $SERVICE_NAME $PORT; do
+              echo "dev-portal is not ready yet. Waiting..."
+              sleep 1
+            done
```

Comment:
I believe this while loop infinitely runs if the port is not going to be ready. so, it'd be better if we have some timeout mechanism here

Reply:
or maybe, what do you think checking if the pod is ready by using `kubectl`?

Reply:
The while loop only runs until it is able to connect to the developer portal pod. The dev portal takes a lot of time to start, so it keeps pinging it until it gets a response and sends a request to the portal.

Reply:
what happens if the bootstrap pod does not run due to some error? Then, this job might stuck. I would suggest adding some timeout mechanism here. What do you think? A simple t/o mechanism might be sufficient like:

```bash
count=0
while ! nc -z $SERVICE_NAME $PORT; do
              echo "dev-portal is not ready yet. Waiting..."
              if [ $count -eq $max_count]; then # it might be some number to accommodate 10 minutes
                   echo "Exceeded maximum count"
              fi
              count=$((count + 1))

              sleep 5
done

```

Reply:
yeah. A timeout works.